### ISSUE TITLE
Maximum call stack size exceeded error after upgrading to 0.20.4

### ISSUE THREAD

**Issue by patryk0605** [recharts/recharts#459]
Using the following chart: 

```jsx
<ResponsiveContainer width="100%" height="100%">
  <RadialBarChart cx="50%" cy="100%" innerRadius="1%" outerRadius="100%" data={data} onClick={::this.requestTagsWithPattern} >
    <Legend align="left" verticalAlign="top" onClick={::this.requestTagsWithPattern} />
    <RadialBar label={radialLabelStyle} className="clickable" minAngle={15} background clockWise={true} dataKey="uv" />
    <Tooltip />
  </RadialBarChart>
</ResponsiveContainer>
```

I am receiving the following error when hovering over the `RadialBar`: 

```
util.js:51 Uncaught RangeError: Maximum call stack size exceeded[Symbol.replace] 
@ (program):665replace 
@ (program):119generatePrefixStyle 
@ util.js:51(anonymous function) 
@ util.js:119reduce 
@ (program):999translateStyle 
@ util.js:118render 
@ Tooltip.js:220(anonymous function) 
@ ReactCompositeComponent.js:796measureLifeCyclePerf 
@ ReactCompositeComponent.js:75_renderValidatedComponentWithoutOwnerOrContext 
@ ReactCompositeComponent.js:795
```

This issue started occurring after upgrading from **0.20.2** to **0.20.4**.

**Comment by patryk0605** [comment]
Also, this error only occurs when the `RadialBarChart` has 4 or more values. Hovering over the first 4 values will not cause an issue, but once you try to hover over the 4th or 5th, the chart and tooltip freeze up and fire off the above error in the console.

**Comment by loris** [comment]
+1, do confirm the issue, it also occurs with `PieChart` (whatever the number of displayed values is). By the way, it only occurs when chrome console / react devtool are opened (looks like it comes from some dev-only benchmark methods like `measureLifeCyclePerf`)

**Comment by patryk0605** [comment]
@loris You're right that it's only happening in Chrome as I tried on IE11 with no issue. However it happens regardless of console being open or not - at least for me anyway on `Chrome 53.0.2785.116 m`

**Comment by xile611** [comment]
@patryk0605 I found the problem in 0.20.3, so I try to fix it in 0.20.4. I can't reproduce the problem now.

**Comment by patryk0605** [comment]
@xile611 Please refer to this [fiddle](https://jsfiddle.net/patryk0605/hbyfdkk2/1/) and hover over some of the bars to trigger error.

**Comment by xile611** [comment]
The problem seems to be caused when the [width and height of tooltip change](https://github.com/recharts/recharts/blob/v0.20.4/src/component/Tooltip.js#L111) alternately very small in some cases.

<img width="971" alt="2017-01-17 11 38 58" src="https://cloud.githubusercontent.com/assets/6111424/22027928/63fc0bf0-dd10-11e6-8bab-1900fa6544fb.png">

So I [add a tolerance value](https://github.com/recharts/recharts/blob/master/src/component/Tooltip.js#L113) to decide whether update the state of tooltip.

Please check again.
@patryk0605 @loris

**Comment by patryk0605** [comment]
@xile611 After updating to **0.20.5**, I am no longer encountering the issue. Thanks for the quick turnaround on this.

**Comment by xile611** [comment]
Fixed in recharts@0.20.5.
